feature/experimental/bgp/ate_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -359,7 +428,8 @@ func TestParameters(t *testing.T) {
 			tc.ateConf.Push(t)
 			tc.ateConf.StartProtocols(t)
 			t.Log("Verify BGP session state : ESTABLISHED")
-			gnmi.Await(t, dut, nbrPath.SessionState().State(), time.Second*100, oc.Bgp_Neighbor_SessionState_ESTABLISHED)
+			time.Sleep(time.Second * 5)
```

Comment:
@liulk since there is already a `gnmi.Await` below, is `time.Sleep()` here needed? Thanks.

Reply:
I had a similar comment earlier: https://github.com/openconfig/featureprofiles/pull/787/files/cdc08fb4610aaab9c52eae1d3891de836db45226#r1040170605

I think this is very subtle. If I understand the code correctly, Await will fatal immediately on subscription error which can happen if the path has not become valid.

https://github.com/openconfig/ygnmi/blob/a492ddad11462bfe95df4b6ddda2c0f6453e1168/ygnmi/ygnmi.go#L306-L310

@DanG100 I think the sleep + await is an anti-pattern. Is there a way we can just await with implicit subscription retries?

@xw-g @cprabha I think it's fine to do sleep + await for now.

Reply:
If the path is valid and supported by the target, it MUST not return an error -- rather no data is received. Thus the stream should not be closed, and the sleep is therefore unnecessary if this behaviour is followed.

https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#3524-subscriberesponse-
behavior-table

Reply:
nbrPath is keyed by ateIP: `/network-instances/network-instance[name=DEFAULT]/protocols/protocol[identifier=BGP]/bgp/neighbors/neighbor[neighbor-address=ateIP]`

Are you saying that even though ateIP is missing, the path is still syntactically correct, so the device should just return no value?